api/jobs/gears.py

```diff
@@ -142,11 +160,21 @@ def insert_gear(doc):
         })
         return job.insert()
 
-def remove_gear(name):
-    config.db.gears.remove({'gear.name': name})
+
+def remove_gear(_id):
+    config.db.gears.remove({"_id": _id})
```

Comment:
This is [deprecated](http://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.remove), you probably want `delete_one` instead.